controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/DeleteStreamTask.java

```diff
@@ -53,7 +53,11 @@ public DeleteStreamTask(final StreamMetadataTasks streamMetadataTasks,
         String scope = request.getScope();
         String stream = request.getStream();
         long requestId = request.getRequestId();
-        return streamMetadataStore.isSealed(scope, stream, context, executor)
+
+        return streamMetadataStore.getCreationTime(scope, stream, context, executor)
+            .thenAccept(creationTime -> Preconditions.checkArgument(request.getCreationTime() == 0 ||
```

Comment:
I understand that this PR adds a security check to avoid an old `deleteStream` event to take effect on a re-created Stream. My question is, do we need a more extensive test coverage of other scenarios on re-created Streams? What about "read" operations accessing a Stream's metadata? Are they safe/correct when executed during a Stream re-creation? I assume that the metadata of a Stream is atomically updated across state transitions (created -> deleted -> re-created), but I just wanted your feedback on this.

Reply:
I have created issue #3238 to add appropriate tests for "read" operations accessing recreated stream's metadata.